src/promise.js

```diff
@@ -725,6 +734,10 @@ Promise.prototype._settlePromises = function () {
     var bitField = this._bitField;
     var len = BIT_FIELD_READ(LENGTH_MASK);
 
+    if (this.hadronTest !== undefined && _global.hadronTest !== this.hadronTest) {
+        throw new Error("settling promise created by: " + this.hadronTest + ", inside:" + _global.hadronTest);
```

Comment:
(since people who get this error message probably won't know to look at Bluebird source for more context, it'll be good to give more detail - assuming we keep this in for dev builds)